Spanner/src/SpannerClient.php

```diff
@@ -131,14 +142,18 @@ class SpannerClient
      */
     public function __construct(array $config = [])
     {
+        $emulatorHost = getenv('SPANNER_EMULATOR_HOST');
+
         $this->requireGrpc();
         $config += [
             'scopes' => [
                 self::FULL_CONTROL_SCOPE,
                 self::ADMIN_SCOPE
             ],
             'returnInt64AsObject' => false,
-            'projectIdRequired' => true
+            'projectIdRequired' => true,
+            'hasEmulator' => (bool) $emulatorHost,
+            'emulatorHost' => $emulatorHost
```

Comment:
I don't know what's normal for PHP in terms of testing but are the snippet tests sufficient? Or should we also have unit tests?

Reply:
I believe we've not exhaustively tested the emulator configuration in other services. That's likely the precedent for the manner of tests here. The issue is that the hostname used isn't easily accessible -- it's passed down several layers before actually being set as a property.

@larkee has this been fully tested manually and confirmed to work? Ideally by more than a single person? If so, I'm ok with it as-is. The conventions are pretty well established. I'd like to consider ways in which we can improve the testability of the feature, but that should be done in a holistic manner which looks at Datastore, Firestore and Pubsub as well.

Reply:
Apologies for the delay. I have had another person verify that it works via manual testing. Do you want me to file an issue for improving the testability of the feature?

Reply:
Please do!

Reply:
Done :)